src/reconcile/rec_write.c

```diff
@@ -2417,9 +2420,18 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
         /* We must only have one delta. Building deltas for split case is a future thing. */
         WT_ASSERT(session, last_block);
         multi->block_meta = *block_meta;
+        /* TODO: need to fix checkpoint id assignment for eviction during a checkpoint. */
+        WT_ACQUIRE_READ(checkpoint_id, conn->disaggregated_storage.global_checkpoint_id);
+        if (checkpoint_id != multi->block_meta.checkpoint_id) {
+            WT_ASSERT(session, checkpoint_id > multi->block_meta.checkpoint_id);
+            multi->block_meta.backlink_checkpoint_id = multi->block_meta.checkpoint_id;
+            multi->block_meta.checkpoint_id = checkpoint_id;
+            multi->block_meta.reconciliation_id = 0;
```

Comment:
Is it currently possible for us to reconcile the same original page twice, because we don't rewrite the page in memory after eviction? If so, would this result in us having two different page versions of the same page with the same reconciliation ID (0)? If that's the case, please add a FIXME and create a follow-up ticket. Thanks!

Reply:
No, it will not happen. This case, we will change the checkpoint id so we reset the reconciliation id to 0. The else section is the case we reuse the same checkpoint id and increase the reconciliation id.